src/main/java/com/o19s/es/ltr/ranker/parser/XGBoostJsonParser.java

```diff
@@ -0,0 +1,168 @@
+        private Float leaf;
+        private List<SplitParserState> children;
+
+        public static SplitParserState parse(XContentParser parser, FeatureSet set) {
+            SplitParserState split = PARSER.apply(parser, set);
+            if (split.isSplit()) {
+                if (!split.splitHasAllFields()) {
+                    throw new ParsingException(parser.getTokenLocation(), "This split does not have all the required fields");
+                }
+                if (!split.splitHasValidChildren()) {
+                    throw new ParsingException(parser.getTokenLocation(), "Split structure is invalid, yes, no and/or" +
+                            " missing branches does not point to the proper children.");
+                }
+                if (!set.hasFeature(split.split)) {
+                    throw new ParsingException(parser.getTokenLocation(), "Unknown feature [" + split.split + "]");
+                }
+            } else if (!split.leafHasAllFields()) {
+                throw new ParsingException(parser.getTokenLocation(), "This leaf does not have all the required fields");
+            }
+            return split;
+        }
+        void setNodeId(Integer nodeId) {
+            this.nodeId = nodeId;
+        }
+
+        void setDepth(Integer depth) {
+            this.depth = depth;
+        }
+
+        void setSplit(String split) {
+            this.split = split;
+        }
+
+        void setThreshold(Float threshold) {
+            this.threshold = threshold;
+        }
+
+        void setRightNodeId(Integer rightNodeId) {
+            this.rightNodeId = rightNodeId;
+        }
+
+        void setLeftNodeId(Integer leftNodeId) {
+            this.leftNodeId = leftNodeId;
+        }
+
+        void setMissingNodeId(Integer missingNodeId) {
+            this.missingNodeId = missingNodeId;
+        }
+
+        void setLeaf(Float leaf) {
+            this.leaf = leaf;
+        }
+
+        void setChildren(List<SplitParserState> children) {
+            this.children = children;
+        }
+
+        boolean splitHasAllFields() {
+            return nodeId != null && threshold != null && split != null && leftNodeId != null && rightNodeId != null && depth != null
+                    && children != null && children.size() == 2;
+        }
+
+        boolean leafHasAllFields() {
+            return nodeId != null && leaf != null;
+        }
+
+        boolean splitHasValidChildren() {
+            return children.size() == 2 &&
+                    rightNodeId.equals(children.get(0).nodeId) && leftNodeId.equals(children.get(1).nodeId) &&
+                    (missingNodeId == null || missingNodeId.equals(children.get(1).nodeId));
+        }
+        boolean isSplit() {
+            return leaf == null;
+        }
+
+
+        Node toNode(FeatureSet set) {
+            if (isSplit()) {
+                return new NaiveAdditiveDecisionTree.Split(children.get(1).toNode(set), children.get(0).toNode(set),
```

Comment:
sorry!!
reading back your comments on my fork I completely missed your first line about this issue...
Fixing...
I think I'm confused by the meaning of a successful condition:
for me a condition is met if `feature_score > split.threshold` => `right` but it's `feature_score < split.threshold` => `left`.
So for me `yes` was `go right`...
We don't really care but `missing` => `no` while it would make more sense to go to `yes`?
Well... I'll ignore completely `missing`... I have added some code to constraint the structure but since we don't use it it's confusing to enforce a particular side for `missing`.